src/controllers/ImageTransformsController.php

```diff
@@ -121,15 +121,11 @@ public function actionSave(): ?Response
             $errors = true;
         }
 
-        if (!empty($transform->quality) && (!is_numeric($transform->quality) || $transform->quality > 100 || $transform->quality < 1)) {
+        if ($transform->quality && ($transform->quality > 100 || $transform->quality < 1)) {
             $this->setFailFlash(Craft::t('app', 'Quality must be a number between 1 and 100 (included).'));
             $errors = true;
         }
 
-        if (empty($transform->quality)) {
-            $transform->quality = null;
-        }
-
```

Comment:
At this point we can be sure `$transform->quality` is an int between 1-100, or `null`, so this is not needed.